tests/test_api.py

```diff
@@ -495,6 +497,14 @@ def test_delegation_class(self):
         delegations = Delegations.from_dict(copy.deepcopy(delegations_dict))
         self.assertEqual(delegations_dict, delegations.to_dict())
 
+        # Test some basic missing data
+        invalid_delegations_dicts = [
+            {},
+            {"keys":{}, "roles":[]},
+        ]
+        for d in invalid_delegations_dicts:
+            with self.assertRaises(KeyError):
+                Targets.from_dict(d)
```

Comment:
I'm confused about your intentions here. 
Either` Delegations.from_dict(d)` instead of ` Targets.from_dict(d)` or the invalid input dictionaries should better represent a `Targets` one.

Reply:
Oops, I created the test code in the targets test case and just moved it...